apps/src/netsim/NetSimRouterNode.js

```diff
@@ -290,12 +302,39 @@ NetSimRouterNode.getNodeType = function () {
 };
 
 /**
+ * Makes sure that the given specification contains the fields that this
+ * router needs to do its job.
+ * @param {packetHeaderSpec} packetSpec
+ * @private
+ */
+NetSimRouterNode.prototype.validatePacketSpec_ = function (packetSpec) {
```

Comment:
PacketEncoder has its own validate method; maybe this needs a rename.  A spec that is valid from the encoder's point of view (e.g. `[]`) may not be sufficient for the router's purposes, as the router needs a `toAddress` to do its job.

Of course, all this method is doing is spitting warnings into the log; we could do without it entirely, and be okay when the router drops all packets if it doesn't have a spec for them.